frontend/src/metabase/visualizations/visualizations/CartesianChart/use-models-and-option.ts

```diff
@@ -41,32 +41,45 @@ export function useModelsAndOption({
   );
 
   const chartModel = useMemo(() => {
+    const hasTimelineEvents = timelineEvents
+      ? timelineEvents.length !== 0
+      : false;
+
     switch (card.display) {
       case "waterfall":
         return getWaterfallChartModel(
           seriesToRender,
           settings,
+          width,
+          hasTimelineEvents,
           renderingContext,
         );
       default:
         return getCartesianChartModel(
           seriesToRender,
           settings,
+          width,
```

Comment:
It is [initially planned](https://www.notion.so/metabase/ECharts-tech-design-15a2fc793f164ddb80c9723745c17d0a?pvs=4#fcdd3cfbee40418f8dcd0d33278286e9) that chart model, computation of which is the slowest, do not depend on chart dimensions. I believe we should not change this.
So I'd consider two options:
- Keep duplicated computation since it is cheap compared to computing the chart model, transforming dataset multiple times
- Extract it to he hook level, but do not put into the model

Reply:
Good point, I'll extract it outside of the chart model